xunit.runner.visualstudio.desktop/VsTestRunner.cs

```diff
@@ -445,12 +459,83 @@ void RunTests(IRunContext runContext, IFrameworkHandle frameworkHandle, LoggerHe
 #endif
 
                 var diagnosticSink = new DiagnosticMessageSink(logger, assemblyDisplayName, runInfo.Configuration.DiagnosticMessagesOrDefault);
-                using (var controller = new XunitFrontController(appDomain, assemblyFileName: assemblyFileName, configFileName: null, shadowCopy: shadowCopy, diagnosticMessageSink: MessageSinkAdapter.Wrap(diagnosticSink)))
+                var diagnosticMessageSink = MessageSinkAdapter.Wrap(diagnosticSink);
+                using (var controller = new XunitFrontController(appDomain, assemblyFileName: assemblyFileName, configFileName: null, shadowCopy: shadowCopy, diagnosticMessageSink: diagnosticMessageSink))
                 {
-                    var xunitTestCases = runInfo.TestCases.Select(tc => new { vs = tc, xunit = Deserialize(logger, controller, tc) })
-                                                          .Where(tc => tc.xunit != null)
-                                                          .ToDictionary(tc => tc.xunit, tc => tc.vs);
+                    var testCasesMap = new Dictionary<string, TestCase>();
+                    var testCases = new List<ITestCase>();
+                    if (runInfo.TestCases == null || !runInfo.TestCases.Any())
```

Comment:
This indicates discovery has been delayed for a Run All operation; we already have the controller here. We do the discovery and filtering.